internal/authentication/oidc/clientcredentials.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package oidc
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	"golang.org/x/oauth2"
+	ccreds "golang.org/x/oauth2/clientcredentials"
+
+	"github.com/dapr/kit/logger"
+)
+
+const (
+	oidcScopeOpenID = "openid"
```

Comment:
The component should probably pass the scope it wants to use. Other components may have different scopes.

Reply:
`openid` is always required for OIDC, but other scopes can be added with the `Scopes` option.

Reply:
But see my other comment. This package is for OAuth2 after all, and not OIDC. If a component wants the "openid" scope, it can request it.